sdk/storage/Azure.Storage.Common/src/Shared/StorageBearerTokenChallengeAuthorizationPolicy.cs

```diff
@@ -0,0 +1,102 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace Azure.Storage.Shared
+{
+    /// <summary>
+    ///
+    /// </summary>
+    public class StorageBearerTokenChallengeAuthorizationPolicy : BearerTokenAuthenticationPolicy, IBearerTokenChallengeOptions
```

Comment:
I'm not sure why does this implement `IBearerTokenChallangeOptions`.

Could we add some test coverage for this policy in storage test suite? best if we can something that can run in live test suite.

Reply:
I'll add some unit tests. The live tests should already be using it (minus the tenant discovery logic).

Reply:
That is true. I was hoping to add some coverage for the discovery logic.

Reply:
I should be able to do that as well.